src/main/java/com/openshift/client/ConnectionBuilder.java

```diff
@@ -122,29 +120,23 @@ public IOpenShiftConnection create() {
 			// TODO: separate auth strategies in UrlConnectionHttpClient
 			IHttpClient client = createHttpClient(
 					clientId,
-					null,
-					null,
-					null,
-					null,
-					token,
+					authStrategy,
 					serverUrl,
 					timeout,
 					callback,
 					sslCipherExclusionRegex);
-			return getAPIResource(null, null, token, createRestService(client));
+			return getAPIResource(null, null, authStrategy.getToken(), createRestService(client));
```

Comment:
+1 with the assumption that this will disappear once we get rid of the v2 classes. In my view noone should know about the internals of the authorization, the authorization would be the one that knows how to fullfill its aim. So this will go away once we remove all v2 classes